tests/components/ecoforest/test_config_flow.py

```diff
@@ -0,0 +1,115 @@
+"""Test the Ecoforest config flow."""
+from unittest.mock import AsyncMock, patch
+
+from pyecoforest.exceptions import EcoforestAuthenticationRequired
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.ecoforest.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_form(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_device, config
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "pyecoforest.api.EcoforestApi.get",
+        return_value=mock_device,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            config,
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["context"]
+    assert result["context"]["unique_id"] == "1234"
```

Comment:
Just double checked with my youtube tests I think this looks better:
```suggestion
    assert "result" in result
    assert result["result"].unique_id == "1234"
```